File: all-submissions/0007-reverse-integer/solution.java
class Solution {
    public int reverse(int x) {
        int answer = 0;
        while (x != 0) {
            int num = x%10;
            x /= 10;
            if (answer > Integer.MAX_VALUE/10 ) return 0;
            if (answer < Integer.MIN_VALUE/10 ) return 0;
            answer = answer * 10 + num;
        }
        return answer;
    }
}

File: all-submissions/0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        ans= 0 
        sign = 1
        if x < 0: 
            sign = -1
            x = x * sign
        while(x>0):
           temp = x%10
           x = x//10
           ans = ans * 10 + temp
           if (ans > 2**31-1):
            return 0
        return ans* sign
        
